### PR TITLE
Benchmark and worker fixes and enhancements

### PR DESCRIPTION
**Pull request by dave2wave**
netty-all dependency needed update for openjdk11 support due to reflection errors in coordinating with workers
In testing Kafka JMS  (Confluent JMS Client) ensureTopicsAreReady could infinite loop. The benchmark started better creating the producers before the consumers.
In testing Pulsar Fast JMS (Starlight for JMS) we needed more consumers to get the messages per second throughput we wanted.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -43,7 +42,6 @@
 		<module>driver-nats-streaming</module>
 		<module>driver-nsq</module>
 		<module>driver-jms</module>
-		<module>driver-redis</module>
```
Comment (merlimat) [inline pom.xml:46]: Pravega and redid shouldn't be removed from here

File: benchmark-framework/pom.xml [hunk 2]
```diff
@@ -165,6 +165,12 @@
 		</dependency>
 
 		<dependency>
+ 		  <groupId>io.netty</groupId>
+ 		  <artifactId>netty-all</artifactId>
+ 		  <version>4.1.48.Final</version>
```
Comment (merlimat) [inline benchmark-framework/pom.xml:170]: The version should be omitted from here since it's already specified in the parent pom.

File: benchmark-framework/src/main/java/io/openmessaging/benchmark/WorkloadGenerator.java [hunk 3]
```diff
@@ -76,10 +76,11 @@ public TestResult run() throws Exception {
         List<String> topics = worker.createTopics(new TopicsInfo(workload.topics, workload.partitionsPerTopic));
         log.info("Created {} topics in {} ms", topics.size(), timer.elapsedMillis());
 
-        createConsumers(topics);
         createProducers(topics);
+        createConsumers(topics);
```
Comment (merlimat) [inline benchmark-framework/src/main/java/io/openmessaging/benchmark/WorkloadGenerator.java:80]: The sequence of creating consumers before producer was deliberate, since that will trigger the creation of subscriptions ensuring the consumer will not miss the initial data. 

What is the reason for this change?

File: benchmark-framework/src/main/java/io/openmessaging/benchmark/WorkloadGenerator.java [hunk 4]
```diff
@@ -76,10 +76,11 @@ public TestResult run() throws Exception {
...
 
-        ensureTopicsAreReady();
+	// Current versions of Kafka don't need to ensure topics are ready. Just create the producers first
+        // ensureTopicsAreReady();
```
Comment (merlimat) [inline benchmark-framework/src/main/java/io/openmessaging/benchmark/WorkloadGenerator.java:83]: Why commenting this?

File: benchmark-framework/src/main/java/io/openmessaging/benchmark/WorkloadGenerator.java [hunk 5]
```diff
@@ -159,9 +160,13 @@ private void ensureTopicsAreReady() throws IOException {
         // In this case we just publish 1 message and then wait for consumers to receive the data
         worker.probeProducers();
 
-        while (true) {
-            CountersStats stats = worker.getCountersStats();
+	// do not loop forever
```
Comment (merlimat) [inline benchmark-framework/src/main/java/io/openmessaging/benchmark/WorkloadGenerator.java:163]: What is the reason for this change? Also, if the call to this method is now commented.

File: benchmark-framework/src/main/java/io/openmessaging/benchmark/Benchmark.java [hunk 6]
```diff
@@ -64,6 +64,9 @@
                 "--workers-file" }, description = "Path to a YAML file containing the list of workers addresses")
         public File workersFile;
 
+        @Parameter(names = { "-x", "--extra" }, description = "Allocate extra consumer workers")
+        boolean extraConsumers;
```
Comment (merlimat) [inline benchmark-framework/src/main/java/io/openmessaging/benchmark/Benchmark.java:68]: What are the `extraConsumers` used for? It should also be included in the description

File: pom.xml (hunk 1 above)
Comment (dave2wave) [reply pom.xml]: Fixed

File: benchmark-framework/pom.xml (hunk 2 above)
Comment (dave2wave) [reply benchmark-framework/pom.xml]: Done

File: benchmark-framework/src/main/java/io/openmessaging/benchmark/WorkloadGenerator.java (hunk 3 above)
Comment (dave2wave) [reply benchmark-framework/src/main/java/io/openmessaging/benchmark/WorkloadGenerator.java:80]: Is this deliberate for other drivers? I'm going to review my branch to see if it matters.

Regarding initial data isn't the benchmark warming up and then resetting stats? Does the initial data matter and in what scenario?

File: benchmark-framework/src/main/java/io/openmessaging/benchmark/WorkloadGenerator.java (hunk 4 above)
Comment (dave2wave) [reply benchmark-framework/src/main/java/io/openmessaging/benchmark/WorkloadGenerator.java:83]: It did not make a difference for me. I was having some trouble with the routine as it was hit or miss that it would work when testing the Confluent JMS Client

File: benchmark-framework/src/main/java/io/openmessaging/benchmark/WorkloadGenerator.java (hunk 5 above)
Comment (dave2wave) [reply benchmark-framework/src/main/java/io/openmessaging/benchmark/WorkloadGenerator.java:163]: I made this change first as I was getting stuck in an infinite loop waiting for the probeProducers. It looks like often, but not always the Probe must already have been consumed.

File: benchmark-framework/src/main/java/io/openmessaging/benchmark/Benchmark.java (hunk 6 above)
Comment (dave2wave) [reply benchmark-framework/src/main/java/io/openmessaging/benchmark/Benchmark.java:68]: This is for Pulsar JMS testing

File: benchmark-framework/src/main/java/io/openmessaging/benchmark/WorkloadGenerator.java (hunk 3 above)
Comment (dave2wave) [reply benchmark-framework/src/main/java/io/openmessaging/benchmark/WorkloadGenerator.java:80]: Reverting that change is looking good, but it's increased my publish latency mean and median averages.

I'll add a new commit tomorrow.

File: benchmark-framework/src/main/java/io/openmessaging/benchmark/WorkloadGenerator.java (hunk 4 above)
Comment (dave2wave) [reply benchmark-framework/src/main/java/io/openmessaging/benchmark/WorkloadGenerator.java:83]: Reverting that change is looking good. The troubles I had were with the Confluent JMS Client. I will turn this back on in the mentioned commit.

File: benchmark-framework/src/main/java/io/openmessaging/benchmark/WorkloadGenerator.java (hunk 5 above)
Comment (eolivelli) [reply benchmark-framework/src/main/java/io/openmessaging/benchmark/WorkloadGenerator.java:162]: what happens if the countdown ends ?

what about reverting this part ?

Comment (dave2wave) [reply benchmark-framework/src/main/java/io/openmessaging/benchmark/WorkloadGenerator.java]: I reverted this part too.
